homeassistant/components/zwave/node_entity.py

```diff
@@ -192,6 +196,42 @@ def node_changed(self):
 
         self.maybe_schedule_update()
 
+    async def node_renamed(self, update_ids=False):
+        """Rename the node and update any IDs."""
+        self._name = node_name(self.node)
+        # Set the name in the devices. If they're customised
+        # the customisation will not be stored as name and will stick.
+        dev_reg = await get_dev_reg(self.hass)
+        device = dev_reg.async_get_device(
+            identifiers={(DOMAIN, self.node_id), },
+            connections=set())
+        dev_reg.async_update_device(device.id, name=self._name)
+        # update sub-devices too
+        for i in count(2):
+            identifier = (DOMAIN, self.node_id, i)
+            device = dev_reg.async_get_device(
+                identifiers={identifier, },
+                connections=set())
+            if not device:
+                break
+            new_name = "{} ({})".format(self._name, i)
```

Comment:
There should be a constant or helper method to generate sub-device names. Where is the original one set ?

Reply:
Yep, good call.